pkg/kubevirt/machine_server.go

```diff
@@ -100,6 +100,12 @@ func (p *MachinePlugin) DeleteMachine(ctx context.Context, req *driver.DeleteMac
 		return nil, util.PrepareErrorf(err, "Create machine %q failed on DecodeProviderSpecAndSecret", req.Machine.Name)
 	}
 
+	validationErrors := validation.ValidateKubevirtSecret(providerSpec, req.Secret)
+	if validationErrors != nil {
+		err = fmt.Errorf("error while validating ProviderSpec %v", validationErrors)
+		return nil, status.Error(codes.Internal, err.Error())
```

Comment:
Why not use `util.PrepareErrorf` here as well? I think it would be more concise and consistent with the rest of the code.

Reply:
So once the function `ValidateKubevirtSecret` is moved to `DecodeProviderSpecAndSecret`, the error will be wrapped anyway in the `util.PrepareErrorf`.